modules/core/src/main/java/org/apache/fluo/core/util/CuratorUtil.java

```diff
@@ -116,6 +116,15 @@ public static CuratorFramework newCurator(String zookeepers, int timeout, String
     }
   }
 
+  public static boolean pathExist(CuratorFramework curator, String zPath) {
+    try {
+      return curator.checkExists().forPath(zPath) != null
+          && !curator.getChildren().forPath(zPath).isEmpty();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```

Comment:
This is a very overly broad catch statement, and a very generic RTE. Is there any way these can be made more narrow? Like, just catch the checked exceptions that are thrown, and throw a specific RTE, like IllegalStateException or IllegalArgumentException?

Reply:
Absolutely, maybe I should refactor FluoAdminImpl.java that I stole that from also?
https://github.com/apache/fluo/blob/01fc804f5914e6f9f2f4bfed8fc6e3e2701c16dd/modules/core/src/main/java/org/apache/fluo/core/client/FluoAdminImpl.java#L498-L505

Reply:
Sure. I didn't notice that, but it would be useful there also.

Reply:
Preliminary digging. I will be investigating further.
https://github.com/apache/curator/blob/9383aa3e91b1e01743ec2cff20f5fd68a66b83c1/curator-framework/src/main/java/org/apache/curator/framework/imps/GetChildrenBuilderImpl.java#L205-L222

Reply:
It looks like starting curator framework throws exception also:
https://github.com/apache/curator/blob/9383aa3e91b1e01743ec2cff20f5fd68a66b83c1/curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorFrameworkImpl.java#L319-L325